python/nano/src/bigdl/nano/pytorch/torch_nano.py

```diff
@@ -311,3 +311,28 @@ def run(self, *args: Any, **kwargs: Any) -> Any:
         """Only for compatibility, don't use it."""
         # this is a abstract method, so we must implement it
         pass
+
+
+def torch_nano(num_processes: int = 1,
+               use_ipex: bool = False,
+               strategy: str = "subprocess",
+               precision: Union[str, int] = 32,
+               *args, **kwargs):
+
+    def decorator(func):
+
+        def new_func(model, optimizer, train_data_loader, val_data_loader, *inner_args, **inner_kwargs):
```

Comment:
Any suggestions on how to handle the dataloaders here?

what are the "legal" numbers of dataloaders users are allowed to use in `self.setup`?

Reply:
> Any suggestions on how to handle the dataloaders here?

maybe we can receive a `Union[DataLoader|List[DataLoader]]` here

> what are the "legal" numbers of dataloaders users are allowed to use in `self.setup`?

`self.setup` only splits dataloaders into multiple subprocesses using Sampler, so it can handle any number of dataloaders

Reply:
I think one or two dataloaders is the most common cases.

But maybe we can detect all dataloaders automatically here, e.g.
```python
def new_func(model, optimizer, *inner_args, **inner_kwargs):
    class DecoratedTorchNano(TorchNano):
        def train(self):
            _model, _optimizer = self.setup(model, optmizer)
            
            # setup all dataloaders automatically
            inner_args=list(inner_args)
            for idx, arg in enumerate(inner_args):
                if isinstance(arg, DataLoader):
                    inner_args[idx] = self.setup_dataloaders(arg)
            for key, value in inner_kwargs.items():
                if isintance(value, DataLoader):
                    inner_kwargs[key] = self.setup_dataloaders(value)
            
            return func(_model, _optimizer, *inner_args, **inner_kwargs)
```
And we can give a warning if we don't find dataloader in the arguments.

We can even use the this methods to detect model and optimizers automatically, so that we can support any parameter list, as long as model, optimizers and dataloaders are all in parameter list. But it might be too complicated to do so.

Reply:
I think this is a great idea. Let me try this.